### PR TITLE
docs(forms): update API reference for common APIs and template-driven directives

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/forms/src/directives/ng_model_group.ts [hunk 1]
```diff
@@ -22,28 +22,37 @@ export const modelGroupProvider: any = {
...
- * the rest of your form, or if some values in your domain model make more sense to
- * consume together in a nested object.
+ * Use this directive to create a sub-group within a form. Use this directive to
+ * validate a sub-group of your form separately from the rest of your form, or if some
+ * values in your domain model make more sense to consume together in a nested object.
  *
- * Pass in the name you'd like this sub-group to have and it will become the key
- * for the sub-group in the form's full value. You can also export the directive into
+ * Provide a name for the sub-group and it will become the key
+ * for the sub-group in the form's full value. You also export the directive into
```
Comment (kara) [inline packages/forms/src/directives/ng_model_group.ts]: Same here. "can" is necessary, as this is optional.

File: packages/forms/src/directives/default_value_accessor.ts [hunk 2]
```diff
@@ -26,7 +26,9 @@ function _isAndroid(): boolean {
 }
 
 /**
- * Turn this mode on if you want form directives to buffer IME input until compositionend
+ * @description
+ * Provider `InjectionToken` to control if form directives buffer IME input until
+ * the "compositionend" event occurs.
```
Comment (kara) [inline packages/forms/src/directives/default_value_accessor.ts]: > Provide this token to control if form directives buffer IME input until the "compositionend" event occurs.

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/forms/src/directives/ng_form.ts [hunk 3]
```diff
@@ -85,25 +102,33 @@ const resolvedPromise = Promise.resolve(null);
...
-   * **updateOn**: Serves as the default `updateOn` value for all child `NgModels` below it
-   * (unless a child has explicitly set its own value for this in `ngModelOptions`).
-   * Potential values: `'change'` | `'blur'` | `'submit'`
-   *
-   * ```html
-   * <form [ngFormOptions]="{updateOn: 'blur'}">
-   *    <input name="one" ngModel>  <!-- this ngModel will update on blur -->
-   * </form>
-   * ```
+   * **updateOn**: Sets as the default `updateOn` value for all child `NgModels` below it
```
Comment (kara) [inline packages/forms/src/directives/ng_form.ts]:
```suggestion
   * **updateOn**: Sets the default `updateOn` value for all child `NgModels` below it
```

File: packages/forms/src/directives/ng_model.ts [hunk 4]
```diff
@@ -47,55 +47,83 @@ const resolvedPromise = Promise.resolve(null);
...
- * full list of properties directly available in `AbstractControlDirective`.
+ * @see `RadioControlValueAccessor` 
+ * @see `SelectControlValueAccessor`
+ * 
+ * @usageNotes
+ * 
+ * ### Using ngModel on a standalone control
  *
- * The following is an example of a simple standalone control using `ngModel`:
+ * The following examples shows a simple standalone control using `ngModel`:
```
Comment (kara) [inline packages/forms/src/directives/ng_model.ts]:
```suggestion
 * The following examples show a simple standalone control using `ngModel`:
```

File: packages/forms/src/directives/ng_model.ts [hunk 5]
```diff
@@ -110,55 +138,58 @@ export class NgModel extends NgControl implements OnChanges,
...
+   * @description
+   * Tracks the configuration options for this `ngModel` instance.
    *
-   * **name**: An alternative to setting the name attribute on the form control element.
-   * Sometimes, especially with custom form components, the name attribute might be used
-   * as an `@Input` property for a different purpose. In cases like these, you can configure
-   * the `ngModel` name through this option.
+   * **name**: An alternative to setting the name attribute on the form control element. See
+   * the [example](api/forms/NgModel#using-ngmodel-on-a-standalone-control) for using `NgModel`
+   * as a stanaldone control.
```
Comment (kara) [inline packages/forms/src/directives/ng_model.ts]:
```suggestion
   * as a standalone control.
```

File: packages/forms/src/directives/ng_model.ts [hunk 6]
```diff
@@ -186,20 +224,50 @@ export class NgModel extends NgControl implements OnChanges,
...
+               * Returns an array that represents the path from the top-level form to this control.
+               * Each index is the string name of the control on that level.
+               */
               get path(): string[] {
                 return this._parent ? controlPath(this.name, this._parent) : [this.name];
               }
 
+              /**
+               * @description
+               * The top-level directive for this conrol if present, otherwise null.
```
Comment (kara) [inline packages/forms/src/directives/ng_model.ts]:
```suggestion
               * The top-level directive for this control if present, otherwise null.
```

**Review by kara — CHANGES_REQUESTED** [review]
Almost there, just a few more typos

**Comment by brandonroberts** [comment]
@kara latest changes are done.

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**